app/controllers/registrations_controller.rb

```diff
@@ -822,19 +823,69 @@ def create
   end
 
   def create_paypal_order
-    return head :forbidden if Rails.env.production?
+    return head :forbidden if PaypalInterface.paypal_disabled?
 
     @registration = registration_from_params
-    render json: PaypalInterface.create_order(@registration)
+    render json: PaypalInterface.create_order(@registration, params[:total_charge])
   end
 
   def capture_paypal_payment
-    return head :forbidden if Rails.env.production?
+    return head :forbidden if PaypalInterface.paypal_disabled?
 
     @registration = registration_from_params
     @competition = @registration.competition
     order_id = params[:order_id]
 
-    render json: PaypalInterface.capture_payment(@competition, order_id)
+    response = PaypalInterface.capture_payment(@competition, order_id)
+    if response['status'] == 'COMPLETED'
+
+      # TODO: Handle the case where there are multiple captures for a payment
+      # 1) Multiple installments
+      # 2) Some failed, some succeeded
+
+      amount_details = response['purchase_units'][0]['payments']['captures'][0]['amount']
+      currency_code = amount_details['currency_code']
+      amount = PaypalRecord.amount_in_cents(amount_details["value"], currency_code)
+      record = PaypalRecord.find_by(record_id: response["id"])
+
+      # Create a Capture object and link it to the PaypalRecord
+      # NOTE: This assumes there is only ONE capture per order - not a valid long-term assumption
+      capture_from_response = response['purchase_units'][0]['payments']['captures'][0]
+      PaypalRecord.create(
+        record_id: capture_from_response['id'],
+        status: capture_from_response['status'],
+        payload: {}, # TODO: Refactor so that we can actually capture the payload? Perhaps this needs to be called in PaypalInterface?
+        amount_in_cents: capture_from_response['amount']['value'],
+        currency_code: capture_from_response['amount']['currency_code'],
+        record_type: :capture,
+        parent_record: record,
+      )
+
+      # Record the payment
+      @registration.record_payment(
+        amount,
+        currency_code,
+        record, # TODO: Add error handling for the PaypalRecord not being found
+        @registration.user.id,
+      )
+    end
+
+    render json: response
+  end
+
+  def refund_paypal_payment
+    registration = Registration.find(params[:id])
+    paypal_order = RegistrationPayment.find(params[:payment_id]).receipt
+    payment_capture_id = paypal_order.capture_id
+
+    refund = PaypalInterface.issue_refund(registration, payment_capture_id)
+
+    registration.record_refund(
+      refund.amount_in_cents.to_i,
```

Comment:
Shouldn't this already be stored as an integer? That's what is done in the StripeTransaction table

Reply:
What is the `PaypalInterface.issue_refund` method returning? That object being returned has an `amount_in_cents` method and you need to figure out whether that's already an int or not.

Reply:
(should have made this comment on the create_table migration)

It's returning a `PaypalRecord` object of type `:refund` - the `amount_in_cents` is a property of PaypalRecord, currently being created in the db as a string, but I think it should be int instead?

Reply:
Yes, in that case you should adjust your migration to be an integer. We definintely always want to store amounts as "what we think we're sending / what we want to send" in integer form.
If Paypal is dumb enough to actually require strings as amount (which pretty much any finance API guy in the whole wide world will tell you is a terrible idea, but eh... they're PayPal...) then we should include that string representation in the `payload` if at all necessary.